lib/avromatic/model/attribute/union.rb

```diff
@@ -0,0 +1,46 @@
+require 'avromatic/model/attribute_type/union'
+
+module Avromatic
+  module Model
+    module Attribute
+      class Union < Virtus::Attribute
+        primitive Avromatic::Model::AttributeType::Union
+
+        def initialize(*)
+          super
+
+          primitive.types.each do |type|
+            member_attributes << Virtus::Attribute.build(type)
+          end
+        end
+
+        def coerce(input)
+          return input if value_coerced?(input)
+
+          result = nil
+          member_attributes.find do |union_attribute|
```

Comment:
I don't think we ever want the result of `find` since that would be one of the `member_attributes`. But I'll introduce a local variable to only assign to `result` if condition is met.